Sharadha- HGN Questionnaire Dashboard: Frontend Implement Community Members List with Filtering, Sorting, and Search

SharadhaS24 opened the pull request:
# Description
Implemented the frontend with mock members as the data to filter names in ascending and descending order. Filter profiles having scores greater than 7. User can also search by name.

## Related PRS (if any):
This frontend PR is not related to any backend PR.
…

## Main changes explained:
- Created the community.jsx file and community.css stylesheet.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as any user.
5. go to dashboard→ Tasks→ task→…
6. Below is the video for reference.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/62a06475-1eca-497a-bf21-7938e1786884

AnthonyWeathers commented:
I followed the instructions to test the sorting but I am stuck on step 5, as I login on an admin account, and try to find the tasks, on the dashboard, but the tasks I do find lead me to a different page then the sorting shown in the video. I've been searching around in the area of my screenshot while following step 5: "go to dashboard→ Tasks→ task→…" so far but no result. So, could you provide further information on how to navigate to the page being added, after login as I can't seem to find my way to it so far.

![PR#3439 Tasks section on dashboard](https://github.com/user-attachments/assets/66e4c908-4e56-41fb-9956-cfcde2ba84e7)

SharadhaS24 commented:
> 5\. http://localhost:3000/hgnhelp/community

@AnthonyWeathers  Please visit this link to test the PR.

AnthonyWeathers commented:
@SharadhaS24 

I refollowed the steps, and manually entered http://localhost:3000/hgnhelp/community, as well as opened the link from here, but received a page not found error in both cases. Even restarted both servers to double check, and still received not found.

![PR#3439 page not found](https://github.com/user-attachments/assets/ad558e23-4a59-4139-b2ee-20746fbfa1d8)

SharadhaS24 commented:
> 5\. Go to http://localhost:3000/hgnhelp/community

@AnthonyWeathers , Hi Anthony I just tried pulling the branch and it worked perfectly on my end. Did you by chance miss the step of npm install? Not sure what is causing the page not found error

AnthonyWeathers commented:
@SharadhaS24 

I am not sure either, I switched back to the development branch, then checked back into this branch, did a fresh npm install, and ran it locally, cleared the site cache, and logged in as an admin and volunteer and still received the page not found. I am using the development branch for the backend, but I don't think that should be an issue. 

Just to list out what I'm doing;

On the frontend, I check into the branch through "git checkout origin/community-member-list-filtering", then do npm install, and lastly npm run start:local to run locally.

On backend, I use the development branch, do npm install, then do npm run build, and finish with npm start.

Then I clear site cache, login, then attempt to enter the url, and face that issue. I am using the versions listed in the setup app locally document.

Worst case maybe just I can't review this PR due to some other settings or hardware on my end.

SharadhaS24 commented:
> @SharadhaS24
> 
> I am not sure either, I switched back to the development branch, then checked back into this branch, did a fresh npm install, and ran it locally, cleared the site cache, and logged in as an admin and volunteer and still received the page not found. I am using the development branch for the backend, but I don't think that should be an issue.
> 
> Just to list out what I'm doing;
> 
> On the frontend, I check into the branch through "git checkout origin/community-member-list-filtering", then do npm install, and lastly npm run start:local to run locally.
> 
> On backend, I use the development branch, do npm install, then do npm run build, and finish with npm start.
> 
> Then I clear site cache, login, then attempt to enter the url, and face that issue. I am using the versions listed in the setup app locally document.
> 
> Worst case maybe just I can't review this PR due to some other settings or hardware on my end.



> @SharadhaS24
> 
> I am not sure either, I switched back to the development branch, then checked back into this branch, did a fresh npm install, and ran it locally, cleared the site cache, and logged in as an admin and volunteer and still received the page not found. I am using the development branch for the backend, but I don't think that should be an issue.
> 
> Just to list out what I'm doing;
> 
> On the frontend, I check into the branch through "git checkout origin/community-member-list-filtering", then do npm install, and lastly npm run start:local to run locally.
> 
> On backend, I use the development branch, do npm install, then do npm run build, and finish with npm start.
> 
> Then I clear site cache, login, then attempt to enter the url, and face that issue. I am using the versions listed in the setup app locally document.
> 
> Worst case maybe just I can't review this PR due to some other settings or hardware on my end.

Just to let you know what I did in the frontend, 
git fetch origin pull/3439/head:pr-branch-3439
git checkout pr-branch-3439
npm install
npm run start:local
and entered http://localhost:3000/hgnhelp/community url

backend
git checkout development
npm start

one thing maybe you can do to verify is that after npm install you can check if the new files(community.jsx,community.css, and 2 other icon image) that I added  is visible within your src->components->hgnhelp.

AnthonyWeathers commented:
@SharadhaS24 

I followed what you did, and checked my files, and I logged in and tried the url, and received the not found page still. Also added a screenshot showing I did your steps and showing the files are visible for myself. I think a 3rd party is needed to verify whether I have something wrong on my end if the PR does work fully as I should have everything needed in terms of setup.

![PR#3439 on branch](https://github.com/user-attachments/assets/a2476936-a764-4e6b-9255-1b4e606d50fd)
